core-primitives/types/src/parentchain.rs

```diff
@@ -19,6 +19,9 @@
 
 use sp_runtime::{generic::Header as HeaderG, traits::BlakeTwo256, MultiAddress, MultiSignature};
 use sp_std::vec::Vec;
+use codec::{Encode, Decode};
+
+use substrate_api_client::StaticEvent;
```

Comment:
We should use the import from `itc_api_client_types` here, we want to avoid a direct dependency on the api-client, they should all be proxied through that crate for easier maintenance.

Reply:
I tried to do this, but then we get a cyclic dependency. Hence, I suggest reverting the refactoring and putting these definitions here back into the indirect executor.

Reply:
Hmm, ok I see it is not straightforward. I can see now that you did this because you want to prevent depending on the indirect executor in the app-libs directory. A comment on why you moved the definitions would have been very helpful. :)

Reply:
I have left it as is for now, as I need to think a bit more about how we'd want to distribute the type definitions. But am I correct that your branch doesn't compile as is? Maybe I screwed something up in the merge, but I don't think so.

Reply:
I just was making it work and trying to work around the cyclic deps. Any suggestions on a comment?